mixer/pkg/api/grpcServer.go

```diff
@@ -131,6 +207,67 @@ func (s *grpcServer) Check(ctx context.Context, req *mixerpb.CheckRequest) (*mix
 	return resp, err
 }
 
+
+func (s *grpcServerEnvoy) checkEnvoy(ctx context.Context, req *authzGRPC.CheckRequest,
+	protoBag *attribute.EnvoyProtoBag, checkBag *attr.MutableBag) (*authzGRPC.CheckResponse, error) {
+	if err := s.dispatcher.Preprocess(ctx, protoBag, checkBag); err != nil {
+		err = fmt.Errorf("preprocessing attributes failed: %v", err)
+		lg.Errora("Check failed: ", err.Error())
+		return nil, grpc.Errorf(codes.Internal, err.Error())
+	}
+
+	lg.Debug("Dispatching to main adapters after running processors")
+	lg.Debuga("Attribute Bag: \n", checkBag)
+	lg.Debug("Dispatching Check")
+	// snapshot the state after we've called the APAs so that we can reuse it
+	// for every check + quota call.
+	//snapApa := protoBag.Snapshot()
```

Comment:
were the snapshots important in Mixer `Check`?  Do we have simplifying assumptions here that make it unnecessary?

Reply:
In Mixer Check protoBag.Restore(snapApa) is called within code related to the Quotas which I don't include for my ExtAuth Check. These correspond to lines 328 - 365 of the file. I had meant to delete the commented out line when I didn't include that section. That is the only place snapApa would've been used.